### PR TITLE
Add hugin

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hmaarrfk** [comment]
linux needs: https://github.com/conda-forge/enblend-feedstock/pull/1

**Comment by hmaarrfk** [comment]
I don't really understand why this isn't being picked up
```
$ mamba install xorg-libxfixes xorg-libsm python=3.7 pip libtiff libpano13 jpeg libpng openexr exiv2 glew sqlite lcms2 wxwidgets=3.1 perl libglu boost fftw enblend "vigra=1.11.1=*_1017"

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.4.3) supported by @QuantStack

        GitHub:  https://github.com/QuantStack/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████

conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
ramonaoptics/noarch      [====================] (00m:00s) Done
ramonaoptics/linux-64    [====================] (00m:00s) Done

Looking for: ['xorg-libxfixes', 'xorg-libsm', 'python=3.7', 'pip', 'libtiff', 'libpano13', 'jpeg', 'libpng', 'openexr', 'exiv2', 'glew', 'sqlite', 'lcms2', 'wxwidgets=3.1', 'perl', 'libglu', 'boost', 'fftw', 'enblend', 'vigra==1.11.1[build=*_1017]']

Transaction

  Prefix: /home/mark/miniforge3/envs/hugin

  Updating specs:

   - xorg-libxfixes
   - xorg-libsm
   - python==3.7
   - pip
   - libtiff
   - libpano13
   - jpeg
   - libpng
   - openexr
   - exiv2
   - glew
   - sqlite
   - lcms2
   - wxwidgets==3.1
   - perl
   - libglu
   - boost
   - fftw
   - enblend
   - vigra==1.11.1[build=*_1017]


  Package                     Version  Build                Channel                    Size
─────────────────────────────────────────────────────────────────────────────────────────────
  Install:
─────────────────────────────────────────────────────────────────────────────────────────────

  _libgcc_mutex                   0.1  conda_forge          conda-forge/linux-64     Cached
  _openmp_mutex                   4.5  0_gnu                conda-forge/linux-64     Cached
  adwaita-icon-theme           3.36.1  h5728ca3_1           conda-forge/linux-64     Cached
  at-spi2-atk                  2.34.2  h1369247_0           conda-forge/linux-64     Cached
  at-spi2-core                 2.36.0  h1369247_1           conda-forge/linux-64     Cached
  atk-1.0                      2.36.0  haf93ef1_1           conda-forge/linux-64     Cached
  boost                        1.72.0  py37h9de70de_0       conda-forge/linux-64     Cached
  boost-cpp                    1.72.0  h7b93d67_2           conda-forge/linux-64     Cached
  bzip2                         1.0.8  h516909a_2           conda-forge/linux-64     Cached
  ca-certificates           2020.6.20  hecda079_0           conda-forge/linux-64     Cached
  cairo                        1.16.0  h3fc0475_1005        conda-forge/linux-64     Cached
  certifi                   2020.6.20  py37hc8dfbb8_0       conda-forge/linux-64     Cached
  dbus                         1.13.6  he372182_0           conda-forge/linux-64     Cached
  enblend                         4.3  h4d0357a_1           conda-forge/linux-64     Cached
  epoxy                         1.5.4  h516909a_2           conda-forge/linux-64     Cached
  exiv2                        0.27.1  had08079_0           conda-forge/linux-64     Cached
  expat                         2.2.9  he1b5a44_2           conda-forge/linux-64     Cached
  fftw                          3.3.8  nompi_h7f3a6c3_1111  conda-forge/linux-64     Cached
  fontconfig                   2.13.1  h1056068_1002        conda-forge/linux-64     Cached
  freetype                     2.10.2  he06d7ca_0           conda-forge/linux-64     Cached
  fribidi                      1.0.10  h516909a_0           conda-forge/linux-64     Cached
  gdk-pixbuf                   2.38.2  h3f25603_4           conda-forge/linux-64     Cached
  gettext                    0.19.8.1  hc5be6a0_1002        conda-forge/linux-64     Cached
  glew                          2.1.0  he1b5a44_0           conda-forge/linux-64     Cached
  glib                         2.65.0  h6f030ca_0           conda-forge/linux-64     Cached
  gobject-introspection        1.64.1  py37h619baee_1       conda-forge/linux-64     Cached
  graphite2                    1.3.13  he1b5a44_1001        conda-forge/linux-64     Cached
  gsl                             2.6  h294904e_0           conda-forge/linux-64     Cached
  gtk3                        3.24.21  h45fd312_1           conda-forge/linux-64     Cached
  harfbuzz                      2.4.0  hee91db6_5           conda-forge/linux-64     Cached
  hdf5                         1.10.6  nompi_h3c11f04_100   conda-forge/linux-64     Cached
  hicolor-icon-theme             0.17  1                    conda-forge/linux-64     Cached
  icu                            67.1  he1b5a44_0           conda-forge/linux-64     Cached
  ilmbase                       2.5.2  h8b12597_0           conda-forge/linux-64     Cached
  jpeg                             9d  h516909a_0           conda-forge/linux-64     Cached
  krb5                         1.17.1  hfafb76e_1           conda-forge/linux-64     Cached
  lcms2                          2.11  hbd6801e_0           conda-forge/linux-64     Cached
  ld_impl_linux-64               2.34  hc38a660_9           conda-forge/linux-64     Cached
  libblas                       3.8.0  17_openblas          conda-forge/linux-64     Cached
  libcblas                      3.8.0  17_openblas          conda-forge/linux-64     Cached
  libcroco                     0.6.13  h8d621e5_1           conda-forge/linux-64     Cached
  libcups                      2.2.12  hf10b501_1           conda-forge/linux-64       6 MB
  libedit                3.1.20191231  h46ee950_1           conda-forge/linux-64     Cached
  libffi                        3.2.1  he1b5a44_1007        conda-forge/linux-64     Cached
  libgcc-ng                     9.2.0  h24d8f2e_2           conda-forge/linux-64     Cached
  libgfortran-ng                7.5.0  hdf63c60_10          conda-forge/linux-64     Cached
  libglu                        9.0.0  he1b5a44_1001        conda-forge/linux-64     Cached
  libgomp                       9.2.0  h24d8f2e_2           conda-forge/linux-64     Cached
  libiconv                       1.15  h516909a_1006        conda-forge/linux-64     Cached
  liblapack                     3.8.0  17_openblas          conda-forge/linux-64     Cached
  libopenblas                  0.3.10  pthreads_hb3c22a3_4  conda-forge/linux-64     Cached
  libpano13                 2.9.20rc2  h160e566_0           conda-forge/linux-64     Cached
  libpng                       1.6.37  hed695b0_1           conda-forge/linux-64     Cached
  librsvg                      2.49.3  h33a7fed_0           conda-forge/linux-64     Cached
  libstdcxx-ng                  9.2.0  hdf63c60_2           conda-forge/linux-64     Cached
  libtiff                       4.1.0  hc7e4089_6           conda-forge/linux-64     Cached
  libuuid                      2.32.1  h14c3975_1000        conda-forge/linux-64     Cached
  libwebp-base                  1.1.0  h516909a_3           conda-forge/linux-64     Cached
  libxcb                         1.13  h14c3975_1002        conda-forge/linux-64     Cached
  libxml2                      2.9.10  h72b56ed_2           conda-forge/linux-64     Cached
  lz4-c                         1.9.2  he1b5a44_1           conda-forge/linux-64     Cached
  ncurses                         6.2  he1b5a44_1           conda-forge/linux-64     Cached
  numpy                        1.19.1  py37h8960a57_0       conda-forge/linux-64     Cached
  openexr                       2.5.2  he513fc3_0           conda-forge/linux-64     Cached
  openssl                      1.1.1g  h516909a_0           conda-forge/linux-64     Cached
  pango                        1.42.4  h7062337_4           conda-forge/linux-64     Cached
  pcre                           8.44  he1b5a44_0           conda-forge/linux-64     Cached
  perl                         5.26.2  h516909a_1006        conda-forge/linux-64     Cached
  pip                          20.1.1  py_1                 conda-forge/noarch       Cached
  pixman                       0.38.0  h516909a_1003        conda-forge/linux-64     Cached
  pthread-stubs                   0.4  h14c3975_1001        conda-forge/linux-64     Cached
  python                        3.7.8  h425cb1d_0_cpython   conda-forge/linux-64     Cached
  python_abi                      3.7  1_cp37m              conda-forge/linux-64     Cached
  readline                        8.0  he28a2e2_2           conda-forge/linux-64     Cached
  setuptools                   49.2.0  py37hc8dfbb8_0       conda-forge/linux-64     Cached
  sqlite                       3.32.3  hcee41ef_1           conda-forge/linux-64     Cached
  tk                           8.6.10  hed695b0_0           conda-forge/linux-64     Cached
  vigra                        1.11.1  py37ha1f80a5_1017    conda-forge/linux-64     Cached
  wheel                        0.34.2  py37_0               conda-forge/linux-64     Cached
  wxwidgets                     3.1.3  h973b898_0           conda-forge/linux-64     Cached
  xorg-fixesproto                 5.0  h14c3975_1002        conda-forge/linux-64     Cached
  xorg-kbproto                  1.0.7  h14c3975_1002        conda-forge/linux-64     Cached
  xorg-libice                  1.0.10  h516909a_0           conda-forge/linux-64     Cached
  xorg-libsm                    1.2.3  h84519dc_1000        conda-forge/linux-64     Cached
  xorg-libx11                   1.6.9  h516909a_0           conda-forge/linux-64     Cached
  xorg-libxau                   1.0.9  h14c3975_0           conda-forge/linux-64     Cached
  xorg-libxdmcp                 1.1.3  h516909a_0           conda-forge/linux-64     Cached
  xorg-libxext                  1.3.4  h516909a_0           conda-forge/linux-64     Cached
  xorg-libxfixes                5.0.3  h516909a_1004        conda-forge/linux-64     Cached
  xorg-libxrender              0.9.10  h516909a_1002        conda-forge/linux-64     Cached
  xorg-renderproto             0.11.1  h14c3975_1002        conda-forge/linux-64     Cached
  xorg-xextproto                7.3.0  h14c3975_1002        conda-forge/linux-64     Cached
  xorg-xproto                  7.0.31  h14c3975_1007        conda-forge/linux-64     Cached
  xz                            5.2.5  h516909a_1           conda-forge/linux-64     Cached
  zlib                         1.2.11  h516909a_1006        conda-forge/linux-64     Cached
  zstd                          1.4.5  h6597ccf_2           conda-forge/linux-64     Cached

  Summary:

  Install: 96 packages

  Total download: 6 MB

─────────────────────────────────────────────────────────────────────────────────────────────

Finished libcups                              (00m:02s)               6 MB      3 MB/s
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

**Comment by hmaarrfk** [comment]
Even my builds on Azure don't seem to have this problem for linux
https://dev.azure.com/ramonaoptics/feedstock-builds/_build/results?buildId=11550&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=937c195f-508d-5135-dc9f-d4b5730df0f7&l=815

and they pull everything from conda-forge

**Comment by hmaarrfk** [comment]
```
$ mamba install xorg-libxfixes xorg-libsm python=3.7 pip libtiff libpano13 jpeg libpng openexr exiv2 glew sqlite lcms2 wxwidgets=3.1 perl libglu boost=1.70.0 fftw=3 enblend "vigra=1.11.1=*_1017"


Encountered problems while solving.
Problem: package vigra-1.11.1-py37ha1f80a5_1017 requires boost >=1.72.0,<1.72.1.0a0, but none of the providers can be installed
```

File: recipes/hugin/yum_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+mesa-libGL
+mesa-dri-drivers
+libselinux
+libXdamage
+libXxf86vm
+libXext
+
```
Comment (scopatz) [inline recipes/hugin/yum_requirements.txt:7]: Please remove blank lines

File: recipes/hugin/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    # https://github.com/conda/conda/pull/9946
+    # In favor of more time to work on a longer term fix
+    # https://github.com/conda/conda-build/blob/master/tests/test_api_build.py#L240
+    - wxwidgets 3.1
+    - perl
+    - libglu  # [linux]
+    # boost was already migrated for vigra, and the build we
+    # need uses boost 1.72
+    - boost 1.72
+    - boost-cpp 1.72
```
Comment (scopatz) [inline recipes/hugin/meta.yaml:83]: Can these be `>=1.72`?

File: recipes/hugin/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,956 @@
+
```
Comment (scopatz) [inline recipes/hugin/CMakeLists.txt:1]: Did you write this yourself?

Comment (hmaarrfk) [reply recipes/hugin/CMakeLists.txt:1]: patching got really tiring with line endings. I made two small mods. I'll update the "patch"

File: recipes/hugin/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipes/hugin/meta.yaml:83]: i was hoping to remove these eventually.

File: recipes/hugin/yum_requirements.txt (hunk 1 above)
Comment (hmaarrfk) [reply recipes/hugin/yum_requirements.txt]: ok.

File: recipes/hugin/CMakeLists.txt (hunk 3 above)
Comment (scopatz) [reply recipes/hugin/CMakeLists.txt:1]: Ahh OK, I ask because I want to makes sure that it has the proper license/copyright statement at the top then

Comment (hmaarrfk) [reply recipes/hugin/CMakeLists.txt:7]: Hopefully what I have is ok. I should probably use the full GPL language. Will do

**Comment by hmaarrfk** [comment]
@scopatz let me know if this is ok now.

I added the license text at the top of the cmake_lists

File: recipes/hugin/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  name: hugin
+  version: {{ version }}
+
+source:
+  #url: https://pilotfiber.dl.sourceforge.net/project/hugin/hugin/hugin-{{ version }}/hugin-{{ version }}.0.tar.bz2
+  url: https://iweb.dl.sourceforge.net/project/hugin/hugin/hugin-{{ version_major_minor }}/hugin-{{ version }}.tar.bz2
+  sha256: 8ba6bdfea246313f142f17f42e066c6888f51b72e4f8814b5e1c84ff56a95a3e
+  # Line endings frustrate me. So I've just vendored the files but documented the patch.
+  # patches:
+  #   - windows_unbundle.patch
```
Comment (scopatz) [inline recipes/hugin/meta.yaml:14]: please remove comments

File: recipes/hugin/windows_unbundle.patch [hunk 5]
```diff
@@ -0,0 +1,295 @@
+diff -r -u a/CMakeLists.txt b/CMakeLists.txt
```
Comment (scopatz) [inline recipes/hugin/windows_unbundle.patch:1]: Is this patch still needed?

**Comment by scopatz** [comment]
This generally looks pretty good, just a little bit of cleanup?

**Comment by hmaarrfk** [comment]
I would like to keep the patch commented. it makes it really clear what I changed compared to upstream.

**Comment by scopatz** [comment]
Ok sounds good

**Comment by hmaarrfk** [comment]
Hey @scopatz thanks for workign through this. Was there anything else you were interested in improving here at this stage?

**Comment by scopatz** [comment]
Sorry, yeah, this could go in

**Comment by hmaarrfk** [comment]
Thanks! As expected, Boost gave me trouble, but I think i managed to migrate enough!!!

**Pull request merged**